inhibit/inhibit.go

```diff
@@ -161,7 +193,9 @@ type InhibitRule struct {
 	Equal map[model.LabelName]struct{}
 
 	// Cache of alerts matching source labels.
-	scache *store.Alerts
+	icache map[model.Fingerprint]iCacheEntry
```

Comment:
It's hard to tell what performance advantage the `equaLabelsFp` has because all of the benchmarks have the same `equaLabelsFp` (the benchmarks don't currently measure any inhibition rules that use `Equals`).